macros/source_relation.sql

```diff
@@ -0,0 +1,19 @@
+{% macro source_relation() %}
+
+{% if var('union_schemas', none)  %}
+, case
+    {% for schema in var('union_schemas') %}
+    when lower(replace(_dbt_source_relation,'"','')) like '%.{{ schema|lower }}.%' then '{{ schema|lower }}'
```

Comment:
Thanks @DylanBaker the vars config helped me successfully run the union'd models! 

I did notice one issue when running this on BigQuery. It seems this replace functions works for Snowflake, Redshift, and Postgres but not Bigquery (and I don't think Spark?). However, BigQuery uses '`' instead of '"' to separate project.schema.table.

I was only able to get this to work with Redshift, Snowflake, and BigQuery with the below update:
```sql
when lower(replace(_dbt_source_relation,{% if target.type == 'bigquery' %} '`' {% else %} '"' {% endif %},'')) like '%.{{ schema|lower }}.%' then '{{ schema|lower }}'
```

I don't like having the conditional within this logic. Do you know of another more dynamic and clean way to account for BigQuery using backticks instead of quotes?

Reply:
We should be able to just do a replace on both right? Something like:
```sql
when lower(replace(replace(_dbt_source_relation,'"',''),"`")) like '%.{{ schema|lower }}.%' then '{{ schema|lower }}'
 ```
That would clean up both and wouldn't need to be conditional. I don't expect double quotes to be within the backticks in BigQuery, and I don't expect backticks to be within the double quotes on any other database.

Reply:
Yeah I was able to get it to work with the below line!
```sql
when lower(replace(replace(_dbt_source_relation,'"',''),"`",''))) like '%.{{ schema|lower }}.%' then '{{ schema|lower }}'
```
With the double replace here and added to the database part of the macro, I am able to get the unioning to work as intended!